app/src/main/java/com/example/mvvmcodebase/data/repository/CocktailRepository.kt

```diff
@@ -0,0 +1,12 @@
+package com.example.mvvmcodebase.data.repository
+
+import com.example.mvvmcodebase.data.CocktailDataSource
+import com.example.mvvmcodebase.data.GetDataCallback
+import com.example.mvvmcodebase.data.model.Cocktail
+
+class CocktailRepository(private val cocktailDataSource: CocktailDataSource) {
+
+    suspend fun getCocktails(letter: String, callback: GetDataCallback<List<Cocktail>>) {
```

Comment:
vì là hàm suspend nên giá trị trả về của nó sẽ là 1 Defer, nơi chứa dữ liệu được trả về trong tương lai. 
Do đó mình không cần phải khai báo callback để chờ kết quả trả về thành công/thất bại nữa 
```suggestion
    suspend fun getCocktails(letter: String) : List<Cocktail>
```